src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -388,6 +480,42 @@ def add_sample(self, sample: Sample) -> None:
             sample: The sample to be added to the buffer.
         """
         self._buffer.append(sample)
+        if self._source_properties.sampling_start is None:
+            self._source_properties.sampling_start = sample.timestamp
+        self._source_properties.received_samples += 1
```

Comment:
What about:
```
if props.sampling_period_s is None:
   self._source_properties.received_samples += 1
```

Reply:
I was thinking that maybe we want to re-calculate the input sampling rate in the future again, maybe regularly at some defined interval, and maybe we want to take into account the whole history (probably not) or just the last T time. For now I think it makes sense to just keep adding, because the variable counts all the samples received so far, it might be confusing that at some point it stops counting, but in the future we might want to stop or probably even better reset it to adjust the input sampling rate from now and then to make sure we are not sticking to a wrong period forever (if we were unlucky enough to happen to measure it when there was some networking issue or something like that).